src/osp/Resource/Package.h

```diff
@@ -90,121 +71,143 @@ StrViewPair_t decompose_str(std::string_view path, const char delim);
+ */
 class Package
 {
-public:
+    // Used to generate sequential IDs at runtime
+    using resource_id = entt::family<struct resource_type>;
 
+public:
 
+    Package(std::string prefix, std::string packageName);
     Package(Package&& move) = default;
     Package(const Package& copy) = delete;
     Package& operator=(Package const& copy) = delete;
 
-    Package(std::string prefix, std::string packageName);
-
-    //ResourceTable m_resources;
-
-    //virtual Magnum::GL::Mesh* request_mesh(const std::string& path);
-
-    //TypeMap<int>::fish;
-    //std::vector< Resource<Magnum::Trade::ImageData> > g_imageData;
-
-    //constexpr ResourceTable& debug_get_resource_table() { return m_resources; }
-
-//    template<class T>
-//    Resource<T>* debug_add_resource(Resource<T>&& resource);
-
-//    template<class T>
-//    Resource<T>* get_resource(std::string const& path);
-
-//    template<class T>
-//    Resource<T>* get_resource(unsigned resIndex);
-
+    /**
+     * Initialize and add a resource to store in this package.
+     *
+     * @tparam TYPE_T   Type of resource to store
+     *
+     * @param path [in] String used to identify this resource
+     * @param args [in] Arguments to pass to TYPE_T constructor
+     * @return Reference counted dependency to new resource
+     */
     template<class TYPE_T, typename ... ARGS_T>
     DependRes<TYPE_T> add(std::string_view path, ARGS_T&& ... args);
 
+    /**
+     * Get a resource by path identifier
+     *
+     * @tparam TYPE_T   Type of resource to get
+     *
+     * @param path [in] String used to identify the target resource
+     * @return Reference counted dependency to the found resource; empty if not
+     *         found.
+     */
     template<class TYPE_T>
-    DependRes<TYPE_T> get(std::string_view path);
+    DependRes<TYPE_T> get(std::string_view path) noexcept;
+
+    // TODO: function for removing specific resources
 
+    /**
+     * Remove all stored resource of the specified type
+     *
+     * @tparam TYPE_T Type of resource to clear
+     */
     template<class TYPE_T>
-    void clear();
+    void clear() noexcept;
 
+    // Stores maps of resoures of a specified type
+    template<class TYPE_T>
     struct GroupType
     {
-        virtual ~GroupType() = default;
-    };
+        GroupType() noexcept = default;
+        GroupType(GroupType&& move) noexcept = default;
+        GroupType& operator=(GroupType&& move) noexcept = default;
+
+        // Needs to be copyable to store in entt::any, but copy is not allowed
+        // entt experimental branch allows move-only types.
+        GroupType(GroupType const& copy) { assert(0); };
 
-    template<class TYPE_T>
-    struct GroupTypeRes : public GroupType
-    {
-        ~GroupTypeRes() = default;
-        //std::vector<TYPE_T> m_types;
         std::map<std::string, Resource<TYPE_T>, std::less<>> m_resources;
     };
 
-    ResPrefix_t get_prefix() const { return m_prefix; }
```

Comment:
This should probably not be returning by value.

Be liberal in parameters you accept, and exacting in values you return.

Const& is my recommendation.

Also, the function is just a getter. Make it const.